magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala

```diff
@@ -171,11 +171,18 @@ object Registry {
 
   trait RegistryConverters extends RegistryProtocols {
 
+    def getAcronymFromPublisherName(publisherName:Option[String]): Option[String] = {
+      publisherName
+        .map("""[^a-zA-Z\s]""".r.replaceAllIn(_,""))
+        .map("""\s""".r.split(_).map(_.trim.toUpperCase).filter(!List("","AND","THE").contains(_)).map(_.take(1)).mkString)
```

Comment:
"of" too I would think

Reply:
We probably want to keep `OF` in order to be consistent with other search engines (e.g. google).
Tested `Department of the Environment and Energy` on google:
So, if you search  `DOEE` on google, you will get `Department of the Environment and Energy` as the first result.
But if you search `DEE`, you won't get `Department of the Environment and Energy` related result.

Reply:
Yeah but the convention people will be used to is dropping the of - e.g. the Department of Home Affairs is known as DHA, not DOHA, Department of Foreign Affairs and Trade is known as DFAT etc.

Reply:
Make sense~ will updated the list~